### ISSUE TITLE
Output variables incorrect after Elastic IP update (changing instance assignment)

### ISSUE THREAD

**Issue by henrylawson** [hashicorp/terraform#5407]
TLDR - When elastic IP assigns itself to an aws_instance, the Public IP of the aws_instance should refresh and reflect the new Elastic IP, I am noticing that it is showing the Public IP before the Elastic IP was assigned.

I have the following scenario:
1. Public IP of Green and Blue EC2 instances configured as output variables
2. Elastic IP currently assigned to Green EC2
3. Change TF config to assign Elastic IP to Blue EC2
4. Run TF apply
5. Apply is successful, note that the last thing done by TF is update the Elastic IP
6. TF outputs the old public IPs of Blue and Green
7. Run TF apply again
8. TF changes nothing
9. TF outputs the correct public IPs of Blue and Green

First TF apply (after Elastic IP assigned to Blue aws_instance, not the EIP is 51.5.153.91):

```
aws_vpc.default: Refreshing state... (ID: vpc-59517c3d)
aws_iam_role.load_balancer: Refreshing state... (ID: load_balancer)
aws_route53_zone.##REMOVED##: Refreshing state... (ID: Z3RC3E17CNT93K)
aws_key_pair.auth: Refreshing state... (ID: ##REMOVED##_load_balancer)
aws_iam_instance_profile.load_balancer: Refreshing state... (ID: load_balancer)
aws_iam_role_policy.load_balancer_ec2_access: Refreshing state... (ID: load_balancer:load_balancer_ec2_access)
aws_iam_role_policy.load_balancer_log_access: Refreshing state... (ID: load_balancer:load_balancer_log_access)
aws_route53_record.ssl_verification: Refreshing state... (ID: ##REMOVED##_CNAME)
aws_internet_gateway.default: Refreshing state... (ID: igw-3810b25c)
aws_subnet.default: Refreshing state... (ID: subnet-5f7ea475)
aws_security_group.ec2_instance: Refreshing state... (ID: sg-b567c4cd)
aws_security_group.load_balancer: Refreshing state... (ID: sg-b667c4ce)
aws_route.internet_access: Refreshing state... (ID: r-rtb-7ef205191080289494)
aws_instance.load_balancer_blue: Refreshing state... (ID: i-cb8c4348)
aws_instance.load_balancer_green: Refreshing state... (ID: i-996fa81a)
aws_eip.load_balancer: Refreshing state... (ID: eipalloc-955950f1)
aws_route53_record.users: Refreshing state... (ID: Z3RC3E17CNT93K_##REMOVED##_A)
aws_route53_record.root: Refreshing state... (ID: Z3RC3E17CNT93K_##REMOVED##_A)
aws_route53_record.www: Refreshing state... (ID: Z3RC3E17CNT93K_##REMOVED##_A)
aws_eip.load_balancer: Modifying...
  instance: "i-996fa81a" => "i-cb8c4348"
aws_eip.load_balancer: Modifications complete

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.

The state of your infrastructure has been saved to the path
below. This state is required to modify and destroy your
infrastructure, so keep it safe. To inspect the complete state
use the `terraform show` command.

State path: .terraform/terraform.tfstate

Outputs:

  load_balancer_blue_public_ip       = 53.87.152.159
  load_balancer_green_public_ip      = 51.5.153.91
```

Second apply, (no changes made):

```
aws_vpc.default: Refreshing state... (ID: vpc-59517c3d)
aws_iam_role.load_balancer: Refreshing state... (ID: load_balancer)
aws_key_pair.auth: Refreshing state... (ID: ##REMOVED##_load_balancer)
aws_route53_zone.##REMOVED##: Refreshing state... (ID: Z3RC3E17CNT93K)
aws_iam_role_policy.load_balancer_log_access: Refreshing state... (ID: load_balancer:load_balancer_log_access)
aws_iam_instance_profile.load_balancer: Refreshing state... (ID: load_balancer)
aws_iam_role_policy.load_balancer_ec2_access: Refreshing state... (ID: load_balancer:load_balancer_ec2_access)
aws_route53_record.ssl_verification: Refreshing state... (ID: ##REMOVED##_CNAME)
aws_subnet.default: Refreshing state... (ID: subnet-5f7ea475)
aws_internet_gateway.default: Refreshing state... (ID: igw-3810b25c)
aws_security_group.ec2_instance: Refreshing state... (ID: sg-b567c4cd)
aws_security_group.load_balancer: Refreshing state... (ID: sg-b667c4ce)
aws_instance.load_balancer_green: Refreshing state... (ID: i-996fa81a)
aws_instance.load_balancer_blue: Refreshing state... (ID: i-cb8c4348)
aws_route.internet_access: Refreshing state... (ID: r-rtb-7ef205191080289494)
aws_eip.load_balancer: Refreshing state... (ID: eipalloc-955950f1)
aws_route53_record.users: Refreshing state... (ID: Z3RC3E17CNT93K_##REMOVED##_A)
aws_route53_record.www: Refreshing state... (ID: Z3RC3E17CNT93K_##REMOVED##_A)
aws_route53_record.root: Refreshing state... (ID: Z3RC3E17CNT93K_##REMOVED##_A)

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

  load_balancer_blue_public_ip       = 51.5.153.91
  load_balancer_green_public_ip      = 55.208.117.148
```

Note that on the second apply, 0 added, 0 changed, 0 destroyed, however the IP has swapped from green to blue. Ideally this should of been reflected after the first TF `apply`.

**Comment by kbroughton** [comment]
I've seen the same behavior.
It the moment, i do a terraform refresh to resolve it.
